src/Patches/PlayerControllerPatch.cs

```diff
@@ -10,87 +12,116 @@ namespace OoLunar.LethalCompanyPatched.Patches
+        private static readonly FieldInfo? _jumpDelayConfigField  = AccessTools.Field(typeof(LethalCompanyPatchedPlugin), nameof(LethalCompanyPatchedPlugin.JumpDelay));
+        private static readonly FieldInfo? _instantSprintConfigField = AccessTools.Field(typeof(LethalCompanyPatchedPlugin), nameof(LethalCompanyPatchedPlugin.InstantSprint));
+        private static readonly MethodInfo? _configEntryFloatValueMethod = AccessTools.Method(typeof(ConfigEntry<float>), "get_Value");
+        private static readonly MethodInfo? _configEntryBoolValueMethod  = AccessTools.Method(typeof(ConfigEntry<bool>), "get_Value");
+        private static bool _tempCrouch;
+        private static readonly int Crouching = Animator.StringToHash("crouching");
+        private static readonly int StartCrouching = Animator.StringToHash("startCrouching");
+
+        private static int? FindInstruction(List<CodeInstruction> instructions, Predicate<int> predicate, int lookAhead = 0, int lookBehind = 0)
+        {
+            int max_index = instructions.Count - lookAhead;
+            for (int index = lookBehind; index < max_index; index++)
+                if (predicate.Invoke(index))
+                    return index;
+            return null;
+        }
 
-        [HarmonyTranspiler]
-        [HarmonyPatch(typeof(PlayerControllerB), "Update")]
-        public static IEnumerable<CodeInstruction> RemoveJumpDelay(IEnumerable<CodeInstruction> instructions)
+        private static void ReplaceInstruction(
+            List<CodeInstruction> instructions, 
+            List<CodeInstruction> new_instructions,
+            Predicate<int> predicate,
+            int lookAhead = 0,
+            int lookBehind = 0
+            )
         {
-            if (!LethalCompanyPatchedPlugin.InstantJump.Value)
+            
+            if (FindInstruction(instructions, predicate, lookAhead, lookBehind) is { } i)
             {
-                return instructions;
+                instructions.RemoveAt(i);
+                instructions.InsertRange(i, new_instructions);
             }
-
-            List<CodeInstruction> list = new(instructions);
-            for (int i = 0; i < list.Count; i++)
+            else
             {
-                CodeInstruction val = list[i];
-                if (val.opcode != OpCodes.Newobj)
-                {
-                    continue;
-                }
-
-                ConstructorInfo? constructorInfo = val.operand as ConstructorInfo;
-                if (constructorInfo?.DeclaringType == typeof(WaitForSeconds))
-                {
-                    list[i] = new CodeInstruction(OpCodes.Ldnull, null);
-                    list.RemoveAt(i - 1);
-                    i--;
-
-                    LethalCompanyPatchedPlugin.StaticLogger.LogDebug("Patched Instant-Jump");
-                }
+                LethalCompanyPatchedPlugin.StaticLogger.LogError("Failed to replace instruction");
             }
-
+        }
```

Comment:
The git diff did not treat this method nicely... However I would turn this:

```cs
        private static void ReplaceInstruction(List<CodeInstruction> instructions, List<CodeInstruction> new_instructions, Predicate<int> predicate, int lookAhead = 0, int lookBehind = 0)
        {
            if (FindInstruction(instructions, predicate, lookAhead, lookBehind) is { } i)
            {
                instructions.RemoveAt(i);
                instructions.InsertRange(i, new_instructions);
            }
            else
            {
                LethalCompanyPatchedPlugin.StaticLogger.LogError("Failed to replace instruction");
            }
        }
```

Into this:

```cs
        private static void ReplaceInstruction(List<CodeInstruction> instructions, List<CodeInstruction> new_instructions, Predicate<int> predicate, int lookAhead = 0, int lookBehind = 0)
        {
            int? instructionIndex = FindInstruction(instructions, predicate, lookAhead, lookBehind);
            if (instructionIndex is null)
            {
                LethalCompanyPatchedPlugin.StaticLogger.LogError("Failed to replace instruction");
                return;
            }

            instructions.RemoveAt(i);
            instructions.InsertRange(i, new_instructions);
        }
```